From 7cb0df3580d633153429c594db869d2d49cb5818 Mon Sep 17 00:00:00 2001
From: Dylan Yip <dylan.yip@xilinx.com>
Date: Thu, 18 Mar 2021 17:02:58 -0700
Subject: [PATCH 1784/1852] dt: bindings: sound: Add DT bindings for new
 compatible strings

commit 0c4c78a49d6f7c79ac0d8b9aed37348ab2811d49 from
https://github.com/Xilinx/linux-xlnx.git

New compatible strings are needed to properly assign device names for
the sysfs device node. If the old compatible string is used, there is a
warning trace when unloading the pcm driver:
sysfs: cannot create duplicate filename '/bus/platform/devices/fd4a0000.
zynqmp-display:zynqmp_dp_snd_pcm0'

Signed-off-by: Dylan Yip <dylan.yip@xilinx.com>
Acked-by: Varunkumar Allagadapa <varunkumar.allagadapa@xilinx.com>
Acked-by: Jianqiang Chen <jianqiang.chen@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 Documentation/devicetree/bindings/sound/xlnx,dp-snd-pcm.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Documentation/devicetree/bindings/sound/xlnx,dp-snd-pcm.txt b/Documentation/devicetree/bindings/sound/xlnx,dp-snd-pcm.txt
index 303232a2a375..e76f302a0410 100644
--- a/Documentation/devicetree/bindings/sound/xlnx,dp-snd-pcm.txt
+++ b/Documentation/devicetree/bindings/sound/xlnx,dp-snd-pcm.txt
@@ -4,7 +4,7 @@ The DPDMA driver of ZynqMP DisplayPort subsystem is based on DMA engine,
 and the DP PCM driver is based on snd dmaengine helpers.
 
 Required properties:
- - compatible: Should be "xlnx,dp-snd-pcm".
+ - compatible: Should be "xlnx,dp-snd-pcm" or "xlnx,dp-snd-pcm0" or "xlnx,dp-snd-pcm1".
  - dmas: the phandle list of DMA specifiers. The dma channel ID should be one
    of 4 for audio0 channel or 5 for audio1 channel.
  - dma-names: the indentifier strings for DMAs. The value should be "tx".
@@ -12,7 +12,7 @@ Required properties:
 Example:
 
 	xlnx_dp_snd_pcm0: dp_snd_pcm0 {
-		compatible = "xlnx,dp-snd-pcm";
+		compatible = "xlnx,dp-snd-pcm0";
 		dmas = <&xlnx_dpdma 4>;
 		dma-names = "tx";
 	};
-- 
2.31.1

